Assets/Scripts/PlayerCharacter.cs

```diff
@@ -0,0 +1,256 @@
+                GetComponent<Rigidbody2D>().velocity = new Vector2(0, 0);
+            }
+            GetComponent<Rigidbody2D>().velocity = myRigidBody.velocity/deathVector;
+        }
+        GetMovementInput();
+        if (jumpNumber > 0)
+        {
+            speedJumpReducer = 2; // reduces jump height with each subsequent jump.
+        }
+        /*if (Input.GetKey(KeyCode.Space) && jumpCooldown == 0)
+        {
+            Jump();
+        }*/
+        if (jumpCooldown > 0)
+        {
+            jumpCooldown--;
+        }
+        if (myRigidBody.velocity.y == 0)
+        {
+            jumpNumber = 0; //if the character is not moving vertically, the player has taken at least a moment to rest before jumping again.
+        }
+        if (jumpNumber == 0)
+        {
+            speedJumpReducer = 1; //first jump has no vertical distance reducer.
+        }
+        UpdateAnimationParameters();
+
+    }
+
+    private void UpdateAnimationParameters()
+    {
+        grounded = Physics2D.OverlapCircle(groundCheck.position, groundRadius, whatIsGround);
+        anim.SetBool("IsOnGround", grounded);
+        //sets tells animator if character is falling or on the ground.
+        if (myRigidBody.velocity.y < 0 && grounded == false)
+        {
+            falling = true;
+        }
+        else if (grounded==true || myRigidBody.velocity.y <= 0)
+        {
+            falling = false;
+        }
+        anim.SetBool("Falling", falling);
+        anim.SetFloat("Speed", Math.Abs(myRigidBody.velocity.x)); //lets the animator know the speed of player.
+    }
+
+    private void FixedUpdate()
+    {
+        UpdatePhysicsMaterial();
+        if (Time.time - deathTimer >= timeToDeath + 0.5f || deathTimer == 0)
+        {
+            //character cannot use the move function when dying.
+            if (Input.GetKey(KeyCode.D) || Input.GetKey(KeyCode.A))
+            {
+                Move();//lets player move a fixed amount not dependent on frames
+            }
+            //player flips sprite when going opposite direction of prior movement.
+            if (direction > 0 && !facingRight)
+            {
+                Flip();
+            }
+            else if (direction < 0 && facingRight)
+            {
+                Flip();
+            }
+            deathTimer = 0;
+        }
+    }
+
+    private void UpdatePhysicsMaterial()
+    {
+        //changes physics material when moving or stopping.
+        if (Mathf.Abs(direction) > 0)
+        {
+            playerGroundCollider.sharedMaterial = playerMovingPhysicsMaterial;
+        }
+        else
+        {
+            playerGroundCollider.sharedMaterial = playerStoppingPhysicsMaterial;
+        }
+    }
+
+    private void GetMovementInput()
+    {
+        //sets direction of movement from player commands
+        direction = Input.GetAxisRaw("Horizontal");
```

Comment:
Technically this is a bit more than direction unless you are rounding it to 1 or negative 1 every time. It's direction and magnitude (e.g. if you were using analog stick, you could walk more slowly by not fully pushing the stick in a direction)